File: weekend-core/src/main/java/com/github/aly8246/core/exec/Executor.java
package com.github.aly8246.core.exec;

import org.springframework.data.mongodb.core.query.Query;

import java.lang.reflect.Method;
import java.util.List;

public interface Executor {
void insert();

void delete();

int update();

Object select(Query query, Class<?> returnType, String tableName, Method method);

List<Object> selectList(Query query, Class<?> aClass, String tableName, Method method);
}

File: weekend-core/src/main/java/com/github/aly8246/core/executor/Executor.kt
package com.github.aly8246.core.executor

import com.mongodb.client.FindIterable
import org.bson.Document
import java.lang.reflect.Parameter

interface Executor {
    fun select(sql: String): FindIterable<Document>
    fun insert(sql: String, param: MutableMap<Parameter, Any?>): Int
    fun update(sql: String): Int
    fun delete(sql: String): Int
}